### PR TITLE
complete get_service_port_forward fn

### PR DESCRIPTION
**Pull request by Dominion5254**
Complete get_service_port_forward function

### CODE REVIEW

File: core/startos/src/service/service_effect_handler.rs [hunk 1]
```diff
@@ -328,8 +328,23 @@ async fn get_container_ip(context: EffectContext, _: Empty) -> Result<Ipv4Addr,
 async fn get_service_port_forward(
     context: EffectContext,
     data: GetServicePortForwardParams,
-) -> Result<Value, Error> {
-    todo!()
+    host_id: HostId,
```
Comment (dr-bonez) [inline core/startos/src/service/service_effect_handler.rs:331]: this should be part of `GetServicePortForwardParams`

File: core/startos/src/net/forward.rs [hunk 2]
```diff
@@ -86,6 +86,9 @@ impl LanPortForwardController {
 
         update_forward(external, prev, next).await
     }
+    pub fn get_forwards(&self) -> Mutex<BTreeMap<u16, BTreeMap<SocketAddr, Weak<()>>>> {
```
Comment (dr-bonez) [inline core/startos/src/net/forward.rs:89]: delete if no longer necessary

**Review by dr-bonez — CHANGES_REQUESTED** [review]

File: core/startos/src/service/service_effect_handler.rs (hunk 1 above)
Comment (Dominion5254) [reply core/startos/src/service/service_effect_handler.rs]: done

File: core/startos/src/net/forward.rs (hunk 2 above)
Comment (Dominion5254) [reply core/startos/src/net/forward.rs]: deleted

File: core/startos/src/service/service_effect_handler.rs [hunk 3]
```diff
@@ -328,8 +329,22 @@ async fn get_container_ip(context: EffectContext, _: Empty) -> Result<Ipv4Addr,
...
+            match net_service.get_ext_port(data.host_id, internal_port) {
+                Ok(ext_port) => Ok(ext_port),
+                Err(e) => Err(e)
+            }
+        }
+        None => Err(Error::new(
+            eyre!("Upgrade on Weak<ServiceActorSeed> resulted in a None variant"),
+            crate::ErrorKind::NotFound,
+        )),
+    }
```
Comment (dr-bonez) [inline core/startos/src/service/service_effect_handler.rs:347]: `context.deref()` already handles the failed weak deref

your inner match statement is a noop

```suggestion
    let context = context.deref()?;
    let net_service = context.persistent_container.net_service.lock().await;
    net_service.get_ext_port(data.host_id, internal_port)
```

**Review by dr-bonez — CHANGES_REQUESTED** [review]

**Review by dr-bonez — APPROVED** [review]

**Pull request merged**